### ISSUE TITLE
Dependants of coremods aren't loaded with coremods

### ISSUE THREAD

**Issue by mattmess1221** [MinecraftForge/MinecraftForge#4993]
Jars which are defined as `ContainedDeps` are added to the mods list, but there is no association with the coremod jar which uses it. Unless the dep jar contains `TweakClass` or `FMLLoadingPlugin` in the manifest, it will be loaded later as a normal mod or library.

This runs into issues when a mod contains a library like Mixin.

A possible solution would be to have the `CoreModManager` check coremods for contained deps and include those as well.

**Comment by Niko-sk2x** [comment]
As far as I understand it, the idea is that coremods and mods really should be separate jar and coremod shouldn't require the mod, it should just do the bytecode patches.

And this is also why I couldn't get mixin working with jar-in-jar.

The way I tried, was to have mixin library + my mixin classes in one jar, and the actual mod in another jar. That didn't work either, because mixin tried to get bytecode of the classes referenced by mixin classes anyway in some cases (it tried to load even classes other than the interfaces implemented by mixins, I included them in the coremod jar for testing).

And the way I understand it should be done (and I don't really understand it that well at this point, so what I say here may be all wrong), is that Mixin library should define tweak class in it's manifest (or whatever the equivalent will be in 1.13+), and it's jar should be included as a library as is, and the mixin classes should be in yet another jar, which wouldn't be a mod so it would be just added to classpath (is this how it works?). But it would also run into the same issues with mixin trying to use the classes from the main mod jar, so I didn't even attempt that.

And what you want here seems to defy the whole purpose of using the jar in jar system: that with the jar in jar system, coremods simply can't use anything from mod jar. If coremod can make the mod jar load together with the coremod jar, you run into all of the same issues that bundling them in one jar has.
